opentok-whiteboard.js

```diff
@@ -96,90 +93,95 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
 
-            var draw = function (update) {
-                if (!ctx) {
-                    ctx = canvas.getContext("2d");
-                    ctx.lineCap = "round";
-                    ctx.fillStyle = "solid";
-                }
-                if (!update.show)
-                    return;
-                if(update.mode=="pen"){
-                    ctx.globalCompositeOperation = "source-over";
-                    ctx.strokeStyle = update.color;
-                    ctx.lineWidth = update.lineWidth;
-                    ctx.beginPath();
-                    ctx.moveTo(update.fromX, update.fromY);
-                    ctx.lineTo(update.toX, update.toY);
-                    ctx.stroke();
-                    ctx.closePath();
+            var draw = function (update, remote) {
+                if (update.end && scope.path) {
+                    start = drawHistory.length;
+                    drawHistory.push(scope.path);
+                    undoStack.push(scope.path);
+                    count = 0;
+
+                    // Apply smoothing
+                    scope.path.smooth({
+                        type: 'continuous'
+                    });
+                    
+                    // End dragging.
+                    client.dragging = false;
+
+                    $timeout(function() {
+                        scope.path = null;
+                    }, 50);
                 } else {
-                    ctx.globalCompositeOperation = "destination-out";
-                    ctx.lineWidth = update.lineWidth;
-                    ctx.beginPath();
-                    ctx.moveTo(update.fromX, update.fromY);
-                    ctx.lineTo(update.toX, update.toY);
-                    ctx.stroke();
+                    if (remote && !scope.path) {
+                        // Create a new path object
+                        scope.path = new $window.paper.Path();
+
+                        // Apply properties
+                        scope.path.strokeColor = update.color;
+                        scope.path.strokeWidth = update.lineWidth;
+                        scope.path.strokeCap = scope.strokeCap;
+                        scope.path.strokeJoin = scope.strokeJoin;
+
+                        if (update.mode === 'eraser') {
+                            scope.path.blendMode = 'destination-out';
+                        }
+
+                        // Move to start and draw a line from there
+                        start = new $window.paper.Point(update.fromX, update.fromY);
+                        scope.path.moveTo(start);
+                    }
+
+                    scope.path.add(update.toX, update.toY);
                 }
             };
             
             var drawUpdates = function (updates) {
                 updates.forEach(function (update) {
-                    draw(update);
-                    drawHistory.push(update);
+                    draw(update, true);
+                    //drawHistory.push(update);
```

Comment:
Why aren't you appending to the drawHistory anymore? Now when new people join they don't get the image, they only get new updates after that point.